exporter/datadogexporter/traces_exporter.go

```diff
@@ -64,6 +67,7 @@ func newTracesExporter(
 		sourceProvider:   sourceProvider,
 		retrier:          clientutil.NewRetrier(params.Logger, cfg.BackOffConfig, scrubber),
 		metadataReporter: metadataReporter,
+		metricsClient:    datadog.NewMetricClient(params.MeterProvider.Meter("datadogexporter")),
```

Comment:
I think the data race happens because you are doing this too late. I would move this to https://github.com/liustanley/opentelemetry-collector-contrib/blob/ad8600b911f1a856d1e8fd0f33534e252a6b4c45/exporter/datadogexporter/factory.go#L141. Since you don't need to store the metrics client itself, the change should be minimal